### DISCUSSION TITLE
Problem while replacing hardrive

### DISCUSSION THREAD

**Discussion by nasirq** [openzfs/zfs#15973] [Q&A]
I have a zfs pool, which was initially configured as a mirror using two harddrives. One of the harddrive developed problem, so I got a new drive and worked to replace it. My mistake that instead of first marking the drive offline in the pool and then removing it, I just removed it and inserted the new drive and used the zpool replace command. 

Unfortunately the new drive also started showing SMART errors, and errors in the resilver process, so I shutdown the machine and removed the new drive also.

Now, I have another drive, but I am unable to import the pool, I have tried everything and I get the error:

cannot import 'pool1': no such pool or dataset
        Destroy and re-create the pool from
        a backup source.

While further exploring the problem (using zdb), I noticed that somehow the pool converted from a mirror to the stripe, and since now the original second drive is not available, the pool cannot be imported.

Pls help me, as the data is very important.

**Comment by IvanVolosyuk** [comment]
You have probably used:
zpool add ....
instead of
zpool attach ...
Now you have a stripped pool where both the old part of the mirror and the
new drive are required to make it function properly. The way forward is to
physically attach the new device again and do 'zpool remove ...
new_device'. Make sure to check cables and the motherboard controller to be
sure it is not the cause of the problems. What kind of errors are you
getting in smart? Some errors can be caused by a bad cable.
--
Regards,
Ivan

**Reply by nasirq** [reply]
Let me clarify with the sequence of events:

1. When the pool was initially created in the past (I think three years ago), I used the command zpool create pool1 mirror device1 device2
2. Now, when replacing the faulty drive, I did not use the add or attach command but the replace command

Secondly, the errors are not because of the faulty sata cable, as the drive is now making noises consistent with a failed drive.

The zd says:

zdb -e pool1

Configuration for import:
        vdev_children: 2
        version: 5000
        pool_guid: 16645203899737867470
        name: 'pool1'
        state: 0
        vdev_tree:
            type: 'root'
            id: 0
            guid: 16645203899737867470
            children[0]:
                type: 'disk'
                id: 0
                guid: 2331848113211191844
                whole_disk: 1
                metaslab_array: 64
                metaslab_shift: 35
                ashift: 12
                asize: 6001160355840
                is_log: 0
                DTL: 113818
                create_txg: 4
                path: '/dev/disk/by-id/ata-ST6000VN0033-2EE110_ZAD5Y301-part1'
                devid: 'ata-ST6000VN0033-2EE110_ZAD5Y301-part1'
                phys_path: 'pci-0000:00:1f.2-ata-1.0'
            children[1]:
                type: 'missing'
                id: 1
                guid: 0
        load-policy:
            load-request-txg: 18446744073709551615
            load-rewind-policy: 2
zdb: can't open 'pool1': No such file or directory

**Reply by IvanVolosyuk** [reply]
Something doesn't add up. You either had striped pool from the beginning on you covered it later. "zdb -h" should have the answer if you still able to get pool history.
